mailpoet/assets/js/src/automation/automation.tsx

```diff
@@ -1,21 +1,50 @@
 import ReactDOM from 'react-dom';
+import { useCallback, useEffect, useRef, useState } from 'react';
 import { BrowserRouter } from 'react-router-dom';
 import { TopBarWithBeamer } from 'common/top_bar/top_bar';
 import { plusIcon } from 'common/button/icon/plus';
 import { Button, Flex } from '@wordpress/components';
+import apiFetch, { APIFetchOptions } from '@wordpress/api-fetch';
 import { Workflow } from './listing/workflow';
 import { AutomationListing } from './listing';
 import { Onboarding } from './onboarding';
 import {
   CreateEmptyWorkflowButton,
   CreateWorkflowFromTemplateButton,
 } from './testing';
-import { useMutation, useQuery } from './api';
+import { ApiError, initializeApi, useMutation, useQuery } from './api';
 import { WorkflowListingNotices } from './listing/workflow-listing-notices';
+import { Notice } from '../notices/notice';
 import { MailPoet } from '../mailpoet';
 
+let errorHandler;
+
 function Content(): JSX.Element {
   const { data, loading, error } = useQuery<{ data: Workflow[] }>('workflows');
+  const noticesRef = useRef([]);
+  const [notices, setNotices] = useState(noticesRef.current);
+  const addNotice = useCallback(
+    (notice) => {
+      noticesRef.current = [notice, ...noticesRef.current];
+      setNotices(noticesRef.current);
+    },
+    [noticesRef],
+  );
```

Comment:
Using a ref here is probably a code smell. I want to have a consistent way for the component and its children to add notices when certain things occur. If I don't use a ref, I sometimes end up with an old copy of `setNotices` that isn't aware of all the latest notices.

I noticed this with the Undo button for trashed automations. Its callback uses the old version of `addNotice`, so it would add its notice but it wouldn't retain the original one that had the Undo button in the first place.

I hope that makes sense. Like I said, this is probably a code smell and I'm sure there's a cleaner way to handle this.

Reply:
I generally don't know how exactly we use the notices in the listing now, but I think the ideal would be to switch to `@wordpress/notices` and use their store. An example here: https://github.com/mailpoet/mailpoet/blob/2c159b16948de91dbc33f1320aa85a961b95a41c/mailpoet/assets/js/src/automation/editor/api-error-handler.tsx#L4

It doesn't need our store to be set up, we just need to render the notices somewhere, such as https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/automation/editor/components/notices/index.tsx.

But I don't know the current system, really. Can we reasonably easily do that?

As for the `addNotice`, that's weird. If `notices` is a dependency of the `useCallback`, then it should work, I guess. Maybe even safer is to use the prev state: `setNotices((prevState) => [notice, ...prevState])`, that should never mismatch them.

Reply:
I'm switching to using a store, but I'm not sure we can use `@wordpress/notices` without it looking weird because those notices use Gutenberg styles. Please let me know if you still think we should try to use them, though.